src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_OF_FIGURES = 5;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random rand = new Random();
+
+    public Figure getRandomFigure() {
+        int figureIndex = rand.nextInt(NUMBER_OF_FIGURES);
+        String color = colorSupplier.getRandomColor();
+        switch (figureIndex) {
+            case 0:
+                return new Square(rand.nextDouble(100) + 1, color);
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. (Checklist point: Creating a figure, don't pass expressions in the constructor.)